crates/core/src/formatter.rs

```diff
@@ -862,3 +865,126 @@ Refunded: {:.10} ETH
         l2_gas_price_fmt = format_gwei(l2_gas_price.into())
     );
 }
+
+/// Prints halt and revert execution errors.
+pub fn print_execution_error<E>(error: &E, tx: Option<&L2Tx>)
```

Comment:
It looks like we can decompose this function on several parts to do the following:

1. Serialize the transaction details (the template with docs, unknown reasons etc. does not depend on this kind of errors that relate to transactions).

```

    | Transaction details:
    |   Transaction Type: EIP1559Transaction
    |   Nonce: 0
    |   To: 0x36615cf349d7f6344891b1e7ca7c72883f5dc049
    |   From: 0xfac041bcf2c4b43319c2c0a39aba53f4cbe44fe5
    |   Gas Limit: 0
    |   Gas Price: 0.04525000 gwei
    |   Gas Per Pubdata Limit: 50000
 ```
2. Forging an elaborated error message
3. Output


I do not have a strong opinion that it should be done as a part of this PR though.